src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolutionInfo.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.Rpc.Configuration
+{
+    internal record struct WorkerConfigurationResolutionInfo(string WorkersDirectoryPath)
+    {
+        public readonly string WorkersDirPath => WorkersDirectoryPath;
```

Comment:
Consider adding the collection of worker directory paths also as a property to this type and keep a single method in the `IWorkerConfigurationResolver` interface as well to return that. It makes sense to encapsulate both piece of information to single immutable type.

Reply:
If I take your example, `BasePath` will be used as input to determine the output `ConfigPaths`. Hence, they can be part of different structures but not the same structure. One is input to method `GetWorkerConfigPaths()` and another is output of this method.

```csharp
internal sealed record WorkerConfigurationInfo
{
    public required string BasePath { get; init; }

    public required IReadOnlyList<string> ConfigPaths { get; init; }
}
```